src/codecs/hdr/encoder.rs

```diff
@@ -39,20 +82,23 @@ impl<W: Write> HdrEncoder<W> {
             let mut bufb = vec![0; width];
             let mut bufe = vec![0; width];
             let mut rle_buf = vec![0; width];
-            for scanline in data.chunks(width) {
-                for ((((r, g), b), e), &pix) in bufr
+            for _scanline_index in 0..height {
+                assert!(flattened_rgbe_pixels.len() >= width); // may reduce the bound checks
+
+                for (((r, g), b), e) in bufr
                     .iter_mut()
                     .zip(bufg.iter_mut())
                     .zip(bufb.iter_mut())
                     .zip(bufe.iter_mut())
-                    .zip(scanline.iter())
                 {
-                    let cp = to_rgbe8(pix);
+                    let cp = flattened_rgbe_pixels.next().unwrap(); // we know it's here because the length is checked earlier
```

Comment:
Can this be included in the chain of `zip` calls above? Roughly the same as the old version, but passing `flattened_rgbe_pixels` instead of `scanline`

Reply:
this should be possible, maybe this was done to minimize the diff for an easier review. unfortunately i don't have time at the moment :/